### PR TITLE
#20 - front-end-only NGO CRUD view

### PR DESCRIPTION
**Pull request by haroon-ali-dev**
created front-end only page for admin panel NGO list CRUD view for #20

### CODE REVIEW

**Comment by Miladepour** [comment]
The functionality is perfect, but I am thinking about improving the design to make it look better for the user.

What do you think about something like this?
IMG_0200

File: client/src/pages/admin/NGOs.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  const [ngos, setNGOs] = useState([]);
+  const [singleNGO, setSingleNGO] = useState([]);
+  const [showFormModal, setShowFormModal] = useState(false);
+  const [showDeleteModal, setShowDeleteModal] = useState([false, 0]);
+  const [formAction, setFormAction] = useState("");
+
+  useEffect(() => {
+    const getNGOs = () => {
+      setNGOs(fileData);
+    }
```
Comment (Miladepour) [inline client/src/pages/admin/NGOs.js:19]: What do you think about sorting the data by organization names in alphabetical order?

  useEffect(() => {
    const getNGOs = () => {
      setNGOs(fileData.sort((a, b) => a.organization.localeCompare(b.organization)));
    };

File: client/src/pages/admin/components/CreateNGO.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
+import { useForm, useFieldArray } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
```
Comment (Miladepour) [inline client/src/pages/admin/components/CreateNGO.js:7]: Please change to double quotes.

**Comment by haroon-ali-dev** [comment]
> The functionality is perfect, but I am thinking about improving the design to make it look better for the user.
> 
> **What do you think about something like this?** IMG_0200
> 
> Because at least one service selection and one telephone number entry are required, we should allow the user to input both.

@Miladepour each drop down must have a remove button with it because of how react-hook-form works so please show me a sketch that includes that thanks. I'm not sure where to put them in the sketch you've shown me.

**Comment by natyfromwonderland** [comment]
What about something like this? 
IMG_20230315_112534

**Comment by haroon-ali-dev** [comment]
> What about something like this? IMG_20230315_112534

this looks good. I'll try to implement this. Thank you.

File: client/src/pages/admin/components/CreateNGO.js (hunk 2 above)
Comment (haroon-ali-dev) [reply client/src/pages/admin/components/CreateNGO.js]: Done.

**Review by Miladepour — APPROVED** [review]
I have thoroughly checked all of the application's functionalities, and it is working perfectly without any errors. This was one of the most challenging parts of the application, but you have managed it in the best possible way. Thank you, Haroon, for doing such a fantastic job!

**Review by natyfromwonderland — COMMENTED** [review]
Thanks Haroon, great job! a minor comment from me would be changing the title of the modal for editing NGO to 'Edit NGO', as otherwise it's 'create' in both instances? So, not to confuse the user.

**Comment by EneVladMihai** [comment]
Components are a bit lenghty in terms of lines. I think the components would be a bit more readable if the svg definitions were pulled into a separate file e.g. 
```
function BinIcon () {
  return (<svg> ... </svg>)
}

function PenPaperIcon () {
  return (<svg> ... </svg>)
}

**Comment by haroon-ali-dev** [comment]
> Thanks Haroon, great job! a minor comment from me would be changing the title of the modal for editing NGO to 'Edit NGO', as otherwise it's 'create' in both instances? So, not to confuse the user.

@natyfromwonderland
Done.

**Comment by haroon-ali-dev** [comment]
> Components are a bit lenghty in terms of lines. I think the components would be a bit more readable if the svg definitions were pulled into a separate file e.g.
> 
> ```
> function BinIcon () {
>   return (<svg> ... </svg>)
> }
> 
> function PenPaperIcon () {
>   return (<svg> ... </svg>)
> }
> ```

@EneVladMihai 
Done.

**Pull request merged**